torchao/csrc/cuda/tensor_core_tiled_layout/tensor_core_tiled_layout.cu

```diff
@@ -146,121 +155,167 @@ __global__ void _dequantize_int4_kernel(
       v[3] = (pack >> 20) & 0x0000000f;
       v[5] = (pack >> 24) & 0x0000000f;
       v[7] = (pack >> 28) & 0x0000000f;
-      int2* v_i32x2 = reinterpret_cast<int2 *>(v);
+      int2 *v_i32x2 = reinterpret_cast<int2 *>(v);
 
-    #pragma unroll
+#pragma unroll
       for (int i = 0; i < 4; ++i) {
         reinterpret_cast<int2 *>(&pOut[ks[i]])[0] = v_i32x2[i];
       }
     }
   }
 }
 
-// output is [n][k] (int32 dtype)
-// input is [n / 8][k / (InnerKTiles * 16)][32][innerKTiles / 2]
-// scales_and_zeros is [numQGroups][n][2]
-// qGroupSize is 32, 64, 128 or 256
-at::Tensor _dequantize_tensor_core_tiled_layout(
-    const at::Tensor& packed_w,
-    const at::Tensor& scales_and_zeros,
-    int64_t group_size,
-    int64_t innerKTiles)
-{
+at::Tensor _ATH_dequantize_tensor_core_tiled_layout(
+    const AtenTensorHandle packed_w, const AtenTensorHandle scales_and_zeros,
+    int64_t group_size, int64_t innerKTiles) {
 
   constexpr int32_t kNTileSize = 8;
   constexpr int32_t kKTileSize = 16;
 
   c10::cuda::CUDAGuard g(packed_w.device());
 
   // packed_w preconditions
-  TORCH_CHECK(packed_w.dim() == 4);
-  TORCH_CHECK(packed_w.dtype() == at::kInt);
-  TORCH_CHECK(packed_w.is_contiguous());
-  TORCH_CHECK(packed_w.size(2) == 32);
-  TORCH_CHECK(packed_w.size(3) == innerKTiles / 2);
+  int64_t packed_w_dim;
+  aoti_torch_get_dim(packed_w, &packed_w_dim);
+  TORCH_CHECK(packed_w_dim == 4);
+
+  int32_t packed_w_dtype;
+  aoti_torch_get_dtype(packed_w, &packed_w_dtype);
+  TORCH_CHECK(packed_w_dtype == at::kInt);
+
+  // is_contiguous not existent today
+  // TORCH_CHECK(packed_w.is_contiguous());
+
+  int64_t packed_w_dim_2_size;
+  aoti_torch_get_size(packed_w, 2, &packed_w_dim_2_size);
+  TORCH_CHECK(packed_w_dim_2_size == 32);
+
+  int64_t packed_w_dim_3_size;
+  aoti_torch_get_size(packed_w, 3, &packed_w_dim_3_size);
+  TORCH_CHECK(packed_w_dim_3_size == innerKTiles / 2);
   TORCH_CHECK(innerKTiles == 2 || innerKTiles == 4 || innerKTiles == 8);
 
-  auto numQGroups = scales_and_zeros.size(0);
-  int N = packed_w.size(0) * kNTileSize;
-  int K = packed_w.size(1) * innerKTiles * kKTileSize;
+  auto numQGroups;
+  aoti_torch_get_size(scales_and_zeros, 0, &numQGroups);
+
+  int64_t packed_w_dim_0_size;
+  aoti_torch_get_size(packed_w, 0, &packed_w_dim_0_size);
+  int N = packed_w_dim_0_size * kNTileSize;
+
+  int64_t packed_w_dim_1_size;
+  aoti_torch_get_size(packed_w, 1, &packed_w_dim_1_size);
+  int K = packed_w_dim_1_size * innerKTiles * kKTileSize;
 
   // scales_and_zeros preconditions
-  TORCH_CHECK(
-      group_size == 32 || group_size == 64 || group_size == 128 ||
-      group_size == 256);
+  TORCH_CHECK(group_size == 32 || group_size == 64 || group_size == 128 ||
+              group_size == 256);
   TORCH_CHECK(numQGroups == K / group_size);
-  TORCH_CHECK(scales_and_zeros.dim() == 3);
-  TORCH_CHECK(scales_and_zeros.size(1) == N);
-  TORCH_CHECK(scales_and_zeros.size(2) == 2);
+
```

Comment:
i might be a bit out of the loop but have you considered helpers like the below so the code before and after looks similar

```cpp
inline int64_t get_dim(aoti_tensor_t tensor) {
    int64_t dim;
    aoti_torch_get_dim(tensor, &dim);
    return dim;
}

inline int32_t get_dtype(aoti_tensor_t tensor) {
    int32_t dtype;
    aoti_torch_get_dtype(tensor, &dtype);
    return dtype;
}

inline int64_t get_size(aoti_tensor_t tensor, int64_t dim) {
    int64_t size;
    aoti_torch_get_size(tensor, dim, &size);
    return size;
}
```

Reply:
The plan is to eventually have a top level helper C++ API make this easier, but yes the shim.c style API is verbose.